src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useRef, useState } from 'react';
+import styles from './ContextMenu.module.css';
+import { Card } from '../Card/Сard.tsx';
+
+interface ContextMenuProps {
+	items: { label: string; action: () => void }[];
+	children: React.ReactElement<{ onContextMenu: (event: React.MouseEvent<HTMLElement>) => void }>;
+
+}
+
+export function ContextMenu({ children, items }: ContextMenuProps) {
+	const [isVisible, setIsVisible] = useState(false);
+	const [position, setPosition] = useState({ top: 0, left: 0 });
+	const menuRef = useRef<HTMLDivElement>(null);
+
+	const handleContextMenu = (e: React.MouseEvent<HTMLElement>) => {
+		e.preventDefault();
+		const { clientX, clientY } = e;
+		setPosition({ top: clientY, left: clientX });
+		setIsVisible(true);
+	};
+
+	const handleOutsideClick = (e: MouseEvent) => {
+		if (menuRef.current && !menuRef.current.contains(e.target as Node)) {
+			setIsVisible(false);
+		}
+	};
+
+	// Attach event listeners
+	useEffect(() => {
+		document.addEventListener('click', handleOutsideClick);
+		return () => {
+			document.removeEventListener('click', handleOutsideClick);
+		};
+	}, []);
+
+	return (
+		<>
+			{children && React.cloneElement(children, { onContextMenu: handleContextMenu })}
```

Comment:
I didn't get why we need to clone element here. Could you give some light on how it should work?

Reply:
This is how it was suggested on the Internet to pass modified props. In my case, I'm adding the `onContextMenu` function.

Another approach would be to create a React context

Reply:
"Optionally, you can implement a context menu popup that opens when a **user clicks on three dots button** and contains "Edit" and "Delete" menu items."

From the description you need to make it much simplier. You need to show menu list at the top right corner when user click on three dots. In this case you don't need cloning, context and so on.

Reply:
Simplified component